### DISCUSSION TITLE
Convert settings for preprocess and forward on ONNX

### DISCUSSION THREAD

**Discussion by nagaokatakuya** [open-mmlab/mmdeploy#1829] [Q&A]
I would like to know the convert settings for completing inference with ONNX when converting the classification(e.g. resnet18) models to ONNX format using MMDeploy.

My understanding of the design of MMDeploy is
SDK is responsible for preprocess and postprocess, and models converted in various formats 
such as ONNX and OpenVINO are recognized only for forward.

https://mmdeploy.readthedocs.io/en/latest/get_started.html#inference-sdk

i.e.

- preprocess (API + SDK) -> forward (ONNX) -> postprocess (API + SDK)

is. On the other hand, what we want to achieve is to incorporate preprocess into the model as well. 
That is, as the specification of inferene script is below.

- input: image_path, onnx_model_path
- contents:
    - onnx_model load
    - image read
    - preprocess(normalize, crop, 3ch -> 4ch, etc.)
    - forward
- output: logits

To achieve this
I would like to be able to handle 3,4 processes within the ONNX model. 
Regarding postprocess, it is assumed that it will be incorporated in the web API 
that we are planning to implement independently, so I would like to include preprocess and forward in ONNX.

Reading the mmengine preprocess [specification](https://github.com/open-mmlab/mmengine/blob/8d4885cb4325be88eb2ad75510636d281bc73499/mmengine/model/base_model/data_preprocessor.py#L136), I think it is possible because it is written in torch. thank you.

ref: https://mmclassification.readthedocs.io/en/dev-1.x/api/data_process.html#data-preprocessors
